tensorflow/python/distribute/input_lib.py

```diff
@@ -1755,14 +1758,11 @@ def _create_iterators_per_worker(worker_datasets, input_workers,
 def _create_datasets_from_function_with_input_context(input_contexts,
                                                       input_workers,
                                                       dataset_fn,
-                                                      replication_mode):
+                                                      options):
   """Create device datasets per worker given a dataset function."""
   datasets = []
   for i, ctx in enumerate(input_contexts):
-    if replication_mode == InputReplicationMode.PER_WORKER:
-      worker = input_workers.worker_devices[i]
-    else:
-      worker = input_workers._worker_device_pairs[i][1][0]
+    worker = input_workers.worker_devices[i]
```

Comment:
So.. are you finally going with placing the dataset on the host CPU, but only the iterator on the GPU device?